src/generate.ts

```diff
@@ -262,6 +262,11 @@ export function createGeneratePlugin({
             vue,
             context: tscContext,
           }
+
+          options.tsconfigRaw.compilerOptions ??= {}
+          options.tsconfigRaw.compilerOptions.module = 'preserve'
+          options.tsconfigRaw.compilerOptions.moduleResolution = 'bundler'
```

Comment:
I don't think we should overwrite it, since we'll be reading the tsconfig from the user. In some cases, this could lead to unexpected results, as we've seen in previous issues.

Reply:
What are the unexpected results?
FYI I do this in pkgroll and it's been fine: https://github.com/privatenumber/pkgroll/blob/master/src/rollup/configs/dts.ts#L53-L54

Do you prefer we update the tsconfig in the tests instead?

Reply:
This is a bit tricky, see https://stackblitz.com/edit/vitejs-vite-wfsxz95t

<img width="880" height="365" alt="image" src="https://github.com/user-attachments/assets/c4828a51-5fd8-415d-b68c-8c5ddb49c3f6" />

In general, I’d rather not alter the user’s configuration; if they want this behavior, they can enable it themselves.

Reply:
I think this error is technically valid since `vue` has an [export map](https://unpkg.com/vue@3.5.18/package.json) that disallow importing from `vue/jsx-runtime/index.js`. But ok, fair enough. Will revert.